connectivity/FEATURE_BLE/source/generic/GattClientImpl.h

```diff
@@ -222,21 +243,29 @@ class GattClient :
      */
     GattClientShutdownCallbackChain_t shutdownCallChain;
 
-    PalGattClient& _pal_client;
+    PalGattClient &_pal_client;
     ServiceDiscovery::TerminationCallback_t _termination_callback;
-    PalSigningMonitorEventHandler* _signing_event_handler;
-    mutable ProcedureControlBlock* control_blocks;
+    PalSigningMonitorEventHandler *_signing_event_handler;
+    mutable ProcedureControlBlock *control_blocks;
     bool _is_reseting;
 
+    ::ble::GattClient *client = nullptr;
```

Comment:
why would the implementation need a pointer to the interface it's implementing?

Reply:
The class outputs object that reference the `GattClient` interface.

Reply:
Sounds good, maybe fix the resets as well